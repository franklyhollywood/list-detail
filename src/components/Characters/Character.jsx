import { Link } from 'react-router-dom';
import { Center, Button } from '@chakra-ui/react';

export default function Character({ character }) {
  return (
    <>
      <Center>
        <img
          src={character.photoUrl}
          alt={character.name}
          width="500"
          height="600"
        />
      </Center>
      <Center>
        <div>name: {character.name} </div>
        <div>position: {character.position}</div>
        <div>allies: {character.allies}</div>
        <p>
          <Link to="/">
            <Button>BACK</Button>
          </Link>
        </p>
      </Center>
    </>
  );
}
