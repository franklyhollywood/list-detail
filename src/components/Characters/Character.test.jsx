import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Character from './Character.jsx';

it('renders the character view', () => {
  const character = {
    _id: '5cf5679a915ecad153ab68c8',
    allies: 'Royal Earthbender Guards',
    enemies: 'Chin',
    photoUrl:
      'https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441',
  };

  const { container } = render(
    <MemoryRouter>
      <Character character={character} />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
