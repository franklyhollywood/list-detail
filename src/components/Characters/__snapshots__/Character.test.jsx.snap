// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the character view 1`] = `
<div>
  <img
    height="600"
    src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
    width="500"
  />
  <div>
    name: 
     
  </div>
  <div>
    position: 
  </div>
  <div>
    allies: 
    Royal Earthbender Guards
  </div>
</div>
`;
